pkg/server/v3/delta_test.go

```diff
@@ -0,0 +1,413 @@
+				}
+			case <-time.After(1 * time.Second):
+				t.Fatalf("got no response")
+			}
+		})
+	}
+}
+
+func TestDeltaWatchClosed(t *testing.T) {
+	for _, typ := range testTypes {
+		t.Run(typ, func(t *testing.T) {
+			config := makeMockConfigWatcher()
+			config.closeWatch = true
+			s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+			resp := makeMockDeltaStream(t)
+			resp.recv <- &discovery.DeltaDiscoveryRequest{
+				Node:    node,
+				TypeUrl: typ,
+			}
+
+			// Verify that the response fails when the watch is closed
+			if err := s.DeltaAggregatedResources(resp); err == nil {
+				t.Error("DeltaAggregatedResources() => got no error, want watch failed")
+			}
+
+			close(resp.recv)
+		})
+	}
+}
+
+func TestSendDeltaError(t *testing.T) {
+	for _, typ := range testTypes {
+		t.Run(typ, func(t *testing.T) {
+			config := makeMockConfigWatcher()
+			config.deltaResponses = makeDeltaResponses()
+			s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+			// make a request
+			resp := makeMockDeltaStream(t)
+			resp.sendError = true
+			resp.recv <- &discovery.DeltaDiscoveryRequest{
+				Node:    node,
+				TypeUrl: typ,
+			}
+
+			// check that response fails since send returns error
+			if err := s.DeltaAggregatedResources(resp); err == nil {
+				t.Error("DeltaAggregatedResources() => got no error, want send error")
+			}
+
+			close(resp.recv)
+		})
+	}
+}
+
+func TestDeltaAggregatedHandlers(t *testing.T) {
+	config := makeMockConfigWatcher()
+	config.deltaResponses = makeDeltaResponses()
+	resp := makeMockDeltaStream(t)
+
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:    node,
+		TypeUrl: rsrc.ListenerType,
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:    node,
+		TypeUrl: rsrc.ClusterType,
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:                   node,
+		TypeUrl:                rsrc.EndpointType,
+		ResourceNamesSubscribe: []string{clusterName},
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		TypeUrl:                rsrc.RouteType,
+		ResourceNamesSubscribe: []string{routeName},
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		TypeUrl:                rsrc.SecretType,
+		ResourceNamesSubscribe: []string{secretName},
+	}
+
+	s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+	go func() {
+		if err := s.DeltaAggregatedResources(resp); err != nil {
+			t.Errorf("DeltaAggregatedResources() => got %v, want no error", err)
+		}
+	}()
+
+	count := 0
+	for {
+		select {
+		case <-resp.sent:
+			count++
+			if count >= 5 {
+				close(resp.recv)
+				if want := map[string]int{
+					rsrc.EndpointType: 1,
+					rsrc.ClusterType:  1,
+					rsrc.RouteType:    1,
+					rsrc.ListenerType: 1,
+					rsrc.SecretType:   1,
+				}; !reflect.DeepEqual(want, config.deltaCounts) {
+					t.Errorf("watch counts => got %v, want %v", config.deltaCounts, want)
+				}
```

Comment:
Indention here makes this a bit tricky to read, maybe put want into a separate variable?

Reply:
Maybe something like?
```go
				want := map[string]int{rsrc.EndpointType: 1, rsrc.ClusterType: 1, rsrc.RouteType: 1, rsrc.ListenerType: 1, rsrc.SecretType: 1}
				if !reflect.DeepEqual(want, config.deltaCounts) {
					t.Errorf("watch counts => got %v, want %v", config.deltaCounts, want)
				}
```
Or I can keep the long indentation but that does make it a bit hard to read